pkg/common/catalog/builtin.go

```diff
@@ -0,0 +1,146 @@
+package catalog
+
+import (
+	"context"
+	"sync"
+
+	grpc_recovery "github.com/grpc-ecosystem/go-grpc-middleware/recovery"
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/spire/pkg/common/log"
+	"github.com/zeebo/errs"
+	"google.golang.org/grpc"
+)
+
+type BuiltInPlugin struct {
+	Log          logrus.FieldLogger
+	Plugin       Plugin
+	HostServices []HostServiceServer
+}
+
+// LoadBuiltIn loads a builtin plugin.
+func LoadBuiltInPlugin(ctx context.Context, builtIn BuiltInPlugin) (plugin *CatalogPlugin, err error) {
+	if builtIn.Log == nil {
+		builtIn.Log = newDiscardingLogger()
+	}
+
+	pluginClient := builtIn.Plugin.Plugin.PluginClient()
```

Comment:
Heh, i agree it is a little strange. I'll think about it...

Reply:
I noodled on this but would rather keep it as is. It's the only spot with the stuttering and changing these field names would break consistency with other field names. Normal catalog/plugin consumers won't be exposed to this.